src/pages/api/apiUtils/index.ts

```diff
@@ -56,13 +58,11 @@ export const redirectToError = (
     redirectTo(res, '/error');
 };
 
-export const redirectOnFareType = (req: NextApiRequest, res: NextApiResponse): void => {
-    const cookies = new Cookies(req, res);
-    const fareTypeCookie = unescapeAndDecodeCookie(cookies, FARE_TYPE_COOKIE);
-    const { fareType } = JSON.parse(fareTypeCookie);
+export const redirectOnFareType = (req: NextApiRequestWithSession, res: NextApiResponse): void => {
+    const fareTypeAttribute = getSessionAttribute(req, FARE_TYPE_ATTRIBUTE);
 
-    if (fareType) {
-        switch (fareType) {
+    if (isFareType(fareTypeAttribute)) {
```

Comment:
Do we need to check this? If the method is being called to redirect on fareType, could we could infer that there are no errors on it?

Reply:
we should have errors on it as this is being used elsewhere and not on the page. The errors are only valid when we first set the faretype

Reply:
we also need it as it is a wrapper for the type guarding